include/boost/http_proto/impl/serializer.hpp

```diff
@@ -91,21 +88,18 @@ start(
 
 template<
     class Source,
+    class... Args,
     class>
-auto
+Source&
 serializer::
 start(
     message_view_base const& m,
-    Source&& src0) ->
-        typename std::decay<
-            Source>::type&
+    Args&&... args)
```

Comment:
I think we need the static_assert here:
```
static_assert (
    std::is_constructible<Source, Args...> ||
    std::is_constructible<Source, buffered_base::allocator, Args...> );

```

Reply:
or in the other function actually

Reply:
Do you mean `construct_source` functions?
They already check for constructibility in the enable_if check.

Reply:
if both enable_if fails then the compiler tells the user no overload was found. It would be better to use static_assert instead, as this way we can control the error message. And we can leave a good comment near the assert explaining why their type did not meet the requirements.